src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTest.php

```diff
@@ -153,6 +153,14 @@ public function provideMaxSizeExceededTests()
             [1048577, '1Mi', '1048577', '1048576', 'bytes'],
             [1053818, '1Mi', '1029.12', '1024', 'KiB'],
             [1053819, '1Mi', '1.01', '1', 'MiB'],
+
+            // @see FileValidator::factorizeSizes()
+            // If $limit < $coef, $limitAsString can be < 1 with less than 3 decimals.
+            // In this case, we end up displaying an allowed size < 1 (eg: 0.1 MB).
+            // It looks better to keep on factorizing (to display 100 kB for example).
```

Comment:
As the comment at the beginning of the method describes the algorithm, it might be better to put this explanation there too.